proto-lens-protoc/src/Data/ProtoLens/Compiler/Generate.hs

```diff
@@ -579,29 +627,29 @@ rawFieldAccessor f = "Lens.Family2.Unchecked.lens" @@ getter @@ setter
     setter = lambda ["x__", "y__"]
                     $ recUpdate "x__" [fieldUpdate f "y__"]
 
--- | A lens to access a oneof field.
+-- | A lens to access a case of a oneof field.
 --
 -- lens
---   (\ x__ -> case _Foo'bar x__ of
---       Prelude.Just (Foo'Bar'c x__val) -> Prelude.Just x__val
+--   (\ x__ -> case x__ of
+--       Prelude.Just (Foo'c x__val) -> Prelude.Just x__val
 --       otherwise -> Prelude.Nothing)
---   (\ x__ y__ -> x__{_Foo'bar = Prelude.fmap Foo'Bar'c y__})
-oneofFieldAccessor :: OneofFieldInfo -> Exp
-oneofFieldAccessor (OneofFieldInfo consName encName) =
-        "Lens.Family2.Unchecked.lens" @@ getter @@ setter
-      where
-        getter = lambda ["x__"] $
-            case' (var (unQual encName) @@ var "x__")
-                [ alt
-                    (pApp "Prelude.Just" [pApp (unQual consName) [pVar "x__val"]])
-                    ("Prelude.Just" @@ "x__val")
-                , alt
-                    (pVar "_otherwise")
-                    (con "Prelude.Nothing")
-                ]
-        setter = lambda ["x__", "y__"]
-                    $ recUpdate "x__" [fieldUpdate (unQual encName)
-                    $ ("Prelude.fmap" @@ (con $ unQual consName) @@ "y__")]
+--   (\ _ y__ -> fmap Foo'c y__
```

Comment:
Added more to the comment.

The main change is that previously when we had

    data Foo = Foo Foo'Bar
    data Foo'Bar = Case1 Int32 | Case2 Int64

this function generated a `Lens' Foo (Maybe Int64)`, i.e., both accessing the sum type and also accessing its case.  Now it just accesses the case of the sum type (`Lens' (Maybe Foo'Bar)` (Maybe Int64)`) and delegates the former to more generic code.